### PR TITLE
Remove `MESSENGER_TRANSPORT_DSN` from `webapp-pack` and enable one in messenger

### LINKED ISSUE #1117: symfony/webapp-pack defines a MESSENGER_TRANSPORT_DSN
https://github.com/symfony/recipes/blob/b3395a2477b6d58089f92ef3a0d2e58226825a3a/symfony/webapp-pack/1.0/manifest.json#L7

First, this is a bit weird, but more over this conflict with symfony/messenger.

```symfony new --webapp``` generates the following files

```
# In all environments, the following files are loaded if they exist,
# the latter taking precedence over the former:
#
#  * .env                contains default values for the environment variables needed by the app
#  * .env.local          uncommitted file with local overrides
#  * .env.$APP_ENV       committed environment-specific defaults
#  * .env.$APP_ENV.local uncommitted environment-specific overrides
#
# Real environment variables win over .env files.
#
# DO NOT DEFINE PRODUCTION SECRETS IN THIS FILE NOR IN ANY OTHER COMMITTED FILES.
# https://symfony.com/doc/current/configuration/secrets.html
#
# Run "composer dump-env prod" to compile .env files for production use (requires symfony/flex >=1.2).
# https://symfony.com/doc/current/best_practices.html#use-environment-variables-for-infrastructure-configuration

###> symfony/framework-bundle ###
APP_ENV=dev
APP_SECRET=adea3499ae133cf82f1599794754e256
###< symfony/framework-bundle ###

###> symfony/webapp-pack ###
MESSENGER_TRANSPORT_DSN=doctrine://default?auto_setup=0
###< symfony/webapp-pack ###

###> doctrine/doctrine-bundle ###
# Format described at https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html#connecting-using-a-url
# IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
#
# DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
# DATABASE_URL="mysql://app:!ChangeMe!@127.0.0.1:3306/app?serverVersion=8&charset=utf8mb4"
DATABASE_URL="postgresql://app:!ChangeMe!@127.0.0.1:5432/app?serverVersion=14&charset=utf8"
###< doctrine/doctrine-bundle ###

###> symfony/messenger ###
# Choose one of the transports below
# MESSENGER_TRANSPORT_DSN=doctrine://default
# MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages
# MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages
###< symfony/messenger ###

###> symfony/mailer ###
# MAILER_DSN=null://null
###< symfony/mailer ###
```

As you can see, there is two sections where MESSENGER_TRANSPORT_DSN is defined.

Finally, the version uncommented is in `symfony/webapp-pack`.

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  |

fixes #1117

### CODE REVIEW

File: symfony/messenger/6.0/manifest.json [hunk 1]
```diff
@@ -4,9 +4,9 @@
     },
     "env": {
         "#1": "Choose one of the transports below",
-        "#2": "MESSENGER_TRANSPORT_DSN=doctrine://default",
-        "#3": "MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages",
-        "#4": "MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages"
+        "#2": "MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages",
+        "#3": "MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages",
+        "MESSENGER_TRANSPORT_DSN": "doctrine://default"
```
Comment (nicolas-grekas) [inline symfony/messenger/6.0/manifest.json:9]: `?auto_setup=0`?

Comment (nicolas-grekas) [reply symfony/messenger/6.0/manifest.json:9]: because we prefer using migrations to define the table for messages

Comment (lyrixx) [reply symfony/messenger/6.0/manifest.json:9]: But you said you want something that works out of the box. I'm a bit lost

Comment (nicolas-grekas) [reply symfony/messenger/6.0/manifest.json:9]: There is a reason to the current way, but I also get your point. I don't have the final solution :)

Comment (fabpot) [reply symfony/messenger/6.0/manifest.json:9]: Let's do it step by step. This PR with `?auto_setup=0` is already a step in the right direction IMO.
We need to make migrations more front and center I suppose.
Many frameworks do that already (Rails, Laravel, ...).
Maybe we can add a post install instruction?

Comment (lyrixx) [reply symfony/messenger/6.0/manifest.json]: Done 👍🏼

File: symfony/messenger/6.0/post-install.txt [hunk 2]
```diff
@@ -4,8 +4,10 @@
 
   * To send messages to a transport and handle them asynchronously:
 
-    1. Uncomment the <info>MESSENGER_TRANSPORT_DSN</> env var in <comment>.env</>
+    1. Update the <info>MESSENGER_TRANSPORT_DSN</> env var in <comment>.env</> if needed
        and <info>framework.messenger.transports.async</> in <comment>config/packages/messenger.yaml</>;
-    2. Route your message classes to the async transport in <comment>config/packages/messenger.yaml</>.
+    2. (if using doctrine) Generate a doctrine migration  <info>bin/console console doctrine:migration:diff</>
```
Comment (nicolas-grekas) [inline symfony/messenger/6.0/post-install.txt:9]: some typos on this line:
`(if using Doctrine) Generate a Doctrine migration <info>bin/console doctrine:migration:diff</>`

Comment (lyrixx) [reply symfony/messenger/6.0/post-install.txt:9]: Thanks!

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
@lyrixx I think we should also update messenger/5.4, right?

**Comment by fabpot** [comment]
See https://github.com/symfony/recipes/pull/1176

**Pull request merged**